js/common/modules/@arangodb/graph/pregel-test-helpers.js

```diff
@@ -98,13 +118,69 @@ const testPageRankOnGraph = function(vertices, edges, seeded = false) {
     }
     const pr = new PageRank(dampingFactor, vertices.length);
     for (const resultV of result) {
-        if (!amostEquals(resultV.value, pr.doOnePagerankStep(resultV._key, graph), epsilon)) {
-            console.error(`The Pagerank algorithm 
-            did not find a fixed point: it returned ' +
-                    '${resultV.value}, but another test iteration returned ' + 
-                    '${pr.doOnePagerankStep(resultV._key, graph)} for vertex ${JSON.stringify(resultV)}. ' +
-                    'The ranks returned by the algorithm are ${JSON.stringify(result)}.`);
-        }
+        assertAlmostEquals(resultV.value, pr.doOnePagerankStep(resultV._key, graph), epsilon,
+            "The Pagerank algorithm did not find a fixed point",
+            "it returned",
+            "but another test iteration returned",
+            `for vertex ${JSON.stringify(resultV)}. The ranks returned by the algorithm are ${JSON.stringify(result)}.`
+        );
+    }
+};
+
+/**
```

Comment:
Perhaps for a next refactoring: This file is super long. For a new algorithm we could have a separate file (e.g. in a algorithm directory), for HITS there would be testHITSKleinberOnGraph function, class HITS and makeHITSTestSuite function in it.

Reply:
Agree. Also the filename is not as helpful as it suggests to be. (I already have a ticket.)